src/syft/core/store/store_disk.py

```diff
@@ -44,25 +74,37 @@ def __len__(self) -> int:
         return self.db.__len__()
 
     @syft_decorator(typechecking=True)
-    def keys(self) -> List[UID]:
-        return self.db.keys()
+    def keys(self) -> Iterable[UID]:
+        key_strings = self.db.keys()
+        return [UID.from_string(key_string) for key_string in key_strings]
 
     @syft_decorator(typechecking=True)
-    def values(self) -> List[StorableObject]:
-        return self.db.values()
+    def values(self) -> Iterable[StorableObject]:
+        values = []
+        for blob in self.db.values():
+            value = _deserialize(blob=blob, from_bytes=True)
+            values.append(value)
 
-    @syft_decorator(typechecking=True)
-    def __contains__(self, item: UID) -> bool:
-        return item in self.db
-
-    @syft_decorator(typechecking=True)
-    def __setitem__(self, key: UID, value: StorableObject) -> None:
-        self.db[key] = value.serialize()
-        self.db.commit(blocking=False)
+        return values
 
     @syft_decorator(typechecking=True)
+    def __contains__(self, item: UID) -> bool:
+        return str(item.value) in self.db
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def delete(self, key: UID) -> None:
+        try:
+            obj = self.get_object(key=key)
+            if obj is not None:
+                self.db.__delitem__(str(key.value))
```

Comment:
MAYBE: I think we should not use the ```__delitem__``` or any internal methods.
I would vote to go with ```del self.db[str(key.value)]``` and delete the ```__delitem__``` from down bellow.